src/zmq/zmqnotificationinterface.cpp

```diff
@@ -119,17 +109,11 @@ bool CZMQNotificationInterface::Initialize()
 void CZMQNotificationInterface::Shutdown()
 {
     LogPrint(BCLog::ZMQ, "zmq: Shutdown notification interface\n");
-    if (pcontext)
+    for (std::list<CZMQAbstractNotifier*>::iterator i=notifiers.begin(); i!=notifiers.end(); ++i)
```

Comment:
I didn't touch anything beside the C->C++ conversion on purpose, so that this PR can show how that change looks like.  A general cleanup (including these loops) is in #13686.